javascript/ql/src/Declarations/SuspiciousMethodNameDeclaration.qhelp

```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>
+In TypeScript the keywords <code>constructor</code> and <code>new</code> for 
+member declarations are used to declare constructors in classes and interfaces 
+respectively. 
+However, a member declaration with the name <code>new</code> in an interface 
+or <code>constructor</code> in a class, will declare ordinary methods named 
+<code>new</code> or <code>constructor</code> rather than constructors.
+Similarly, the keyword <code>function</code> is used to declare functions in 
+some contexts, however, using the name <code>function</code> for a class 
+or interface member declaration declares a method named <code>function</code>.    
+</p>
+
+</overview>
+<recommendation>
+
+<p>
+Declare classes as classes and not as interfaces.
+Use the keyword <code>constructor</code> to declare constructors in a class,
+use the keyword <code>new</code> to declare constructors inside interfaces, 
+and don't use <code>function</code> when declaring an interface that is a 
+function.
```

Comment:
Yeah, that was not to well described. 
I think I'll change the wording to mention `call signature`. 
Because one should not use `function` when declaring a `call signature` in an interface.